crates/node/src/lib.rs

```diff
@@ -557,6 +578,7 @@ impl Node {
                 self.ctx_manager.save_context(context)?;
             }
 
+            // InMemoryStorage doesn't have .is_empty() or .commit() ?
             if !storage.is_empty() {
                 storage.commit()?;
             }
```

Comment:
after you change `Handler<Execute>` to also return the `Box<dyn Storage>` that was passed in

we'll need to downcast this back to `RuntimeCompatStore`, you can make [`Reflect`](https://github.com/calimero-network/core/blob/bd5d080dc056db3b026d0f89653cdfe5414e3940/crates/primitives/src/reflect.rs#L33) a supertrait of `Storage`

which should let you call

```rust
storage.downcast_box::<RuntimeCompatStore>()
```